vcd/resource_vcd_vapp_vm.go

```diff
@@ -441,7 +445,18 @@ func resourceVcdVAppVmUpdate(d *schema.ResourceData, meta interface{}) error {
 
 		if d.HasChange("cpus") {
 			err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-				task, err := vm.ChangeCPUCount(d.Get("cpus").(int))
+				var task govcd.Task
+				var err error
+				if d.HasChange("cpu_cores") {
+					if d.Get("cpu_cores") != nil {
+						coreCounts := d.Get("cpu_cores").(int)
+						task, err = vm.ChangeCPUCountWithCore(d.Get("cpus").(int), &coreCounts)
+					} else {
+						return resource.RetryableError(fmt.Errorf("missing cpu_core value: %#v", err))
+					}
+				} else {
+					task, err = vm.ChangeCPUCount(d.Get("cpus").(int))
+				}
```

Comment:
Please add a test that changes the CPU cores.

Reply:
resource_vcd_vapp_vm_test.go covers that

Reply:
For a moment I thought I was looking at master, so went and double checked the stash branch - the test only covers `cpu`, there's nothing around `cores`.

Reply:
in  vcd/resource_vcd_vapp_vm_test.go change:
```
  cpus          = 2
  cpu_cores		= 1
```